### PR TITLE
Expose individual small text changes, along with a large aggregate text change with completion items.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/50905

This is useful for VSCode, which cannot simply apply large changes while producing a minimal diff.  We already had a way to shuttle this data along, but we deprecated in lieu of the simpler approach of just having a single change.  Now we provide both so that all hosts can benefit from the data as they see fit.

### CODE REVIEW

**Review by 333fred — APPROVED** [review]
From the consumption side, this looks like about what I was hoping for.

**Comment by CyrusNajmabadi** [comment]
@jasonmalinowski @genlu @Cosifne ptal :)

File: src/Features/Core/Portable/Completion/CompletionChange.cs [hunk 1]
```diff
@@ -37,34 +42,48 @@ public sealed class CompletionChange
...
         public static CompletionChange Create(
             ImmutableArray<TextChange> textChanges,
             int? newPosition = null,
             bool includesCommitCharacter = false)
         {
-            return new CompletionChange(textChanges, newPosition, includesCommitCharacter);
+            return new CompletionChange(textChanges.Single(), textChanges, newPosition, includesCommitCharacter);
+        }
+
+#pragma warning disable RS0027 // Public API with optional parameter(s) should have the most parameters amongst its public overloads
```
Comment (genlu) [inline src/Features/Core/Portable/Completion/CompletionChange.cs:79]: Hmm, maybe we shouldn't warn when the other overload is obsolete.

**Review by genlu — APPROVED** [review]
:shipit:

**Comment by CyrusNajmabadi** [comment]
Thanks!

**Pull request merged**